docs/0001-nixos-render-docs-Support-gfm-style-admonitions.patch

```diff
@@ -0,0 +1,145 @@
+From ddf20fca1e5d3f5669909e825b6b20332887d006 Mon Sep 17 00:00:00 2001
+From: Quentin Boyer <quentin+dev@familleboyer.net>
+Date: Wed, 11 Sep 2024 00:05:47 +0200
+Subject: [PATCH] nixos-render-docs: Support gfm-style admonitions
+
+---
+ nixos_render_docs/md.py  | 35 +++++++++++++++++++++++++++++++++++
+ tests/sample_md.py       |  3 +++
+ tests/test_asciidoc.py   |  6 ++++++
+ tests/test_commonmark.py |  3 +++
+ tests/test_html.py       |  4 ++++
+ tests/test_manpage.py    |  6 ++++++
+ 6 files changed, 57 insertions(+)
+
+diff --git a/nixos_render_docs/md.py b/nixos_render_docs/md.py
+index 894daf9ca9c7..cde874bc5637 100644
+--- a/nixos_render_docs/md.py
++++ b/nixos_render_docs/md.py
+@@ -341,6 +341,40 @@ def _parse_blockattrs(info: str) -> Optional[tuple[AttrBlockKind, Optional[str],
+         return ('figure', id, classes)
+     return None
+ 
++_ALERT_PATTERN = re.compile(r"^\[\!(TIP|NOTE|IMPORTANT|WARNING|CAUTION)\]", re.IGNORECASE)
++
++def _gfm_alert(md: markdown_it.MarkdownIt):
++    def gfm_alert_to_adm(state: markdown_it.rules_core.StateCore):
```

Comment:
This function is large enough that it makes me uncomfortable having it directly in a patch file.

I think it'll end up being a maintenance nightmare if we ever want to work on it.

I'm tempted to suggest extracting it into either a loose file or a nix string variable.

If we did that, we could add a `@PLACEHOLDER@` to the patch, then substitute in the actual function in `postPatch`.

Reply:
Or maybe I just need to re-read the patch on a larger screen and with fresh eyes tomorrow.

Just using a patch is certainly simpler than having a patch, a loose file, _and_ a substitution...

Reply:
(in term of simplicity)